addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -184,114 +216,113 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
-                            }
-                            break;
-                        default:
-                            break;
-                    }
+                if (pairingEnabled) {
+                    addUser(userName);
+                } else if (!authorizeUser(userName)) {
+                    apiServerError(req, out, HueErrorResponse.UNAUTHORIZED, "Not Authorized");
+                    return;
+                }
+
+                if (pathParts.length == 1) {
+                    /**
+                     * /api/{username}
+                     */
+                    apiDataStore(req, resp);
                 } else {
-                    apiServerError(req, resp, HueErrorResponse.NOT_AVAILABLE, "Hue resource not available");
+                    String function = pathParts[1];
+                    if ("lights".equals(function)) {
+                        switch (pathParts.length) {
+                            case 2:
+                                /**
+                                 * /api/{username}/lights
+                                 */
+                                apiLights(req, out);
+                                break;
+                            case 3:
+                                /**
+                                 * /api/{username}/lights/{id}
+                                 */
+                                apiLight(pathParts[2], req, out);
+                                break;
+                            case 4:
+                                /**
+                                 * /api/{username}/lights/{id}/state
+                                 */
+                                apiState(pathParts[2], req, out);
+                                break;
+                            default:
+                                break;
+                        }
+                    } else if ("groups".equals(function)) {
+                        switch (pathParts.length) {
+                            case 2:
+                                /**
+                                 * /api/{username}/group
+                                 */
+                                emptyResponse(req, out);
+                                break;
+                            case 3:
+                                /**
+                                 * /api/{username}/group/{id}
+                                 */
+                                if ("0".equals(pathParts[2])) {
+                                    apiGroupZero(req, out);
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                    } else {
+                        apiServerError(req, out, HueErrorResponse.NOT_AVAILABLE, "Hue resource not available");
+                    }
                 }
             }
+        } catch (IOException e) {
```

Comment:
This feels like a bad way of error handling, I would expect either a change of status or some logging.